activities.el

```diff
@@ -503,6 +514,39 @@ To be called from `kill-emacs-hook'."
 
 ;;;; Functions
 
+(defun activities--map-window-state-leafs (state func)
+  "Call FUNC on all the leaf nodes in window-state STATE.
```

Comment:
I prefer for the first line to say what the function returns.  How about something like, "Return a list of values of having applied FUNC to leaf nodes in STATE."?  Maybe the function could be called `activities--mapcar-window-state-leafs` to additionally imply that it returns a list like `mapcar`.  What do you think?